snapshotter/Makefile

```diff
@@ -12,33 +12,72 @@
 # permissions and limitations under the License.
 
 EXTRAGOARGS?=
+NUMBER_OF_VMS?=
 
 SOURCES := $(shell find . -name '*.go')
 GOMOD := $(shell go env GOMOD)
 GOSUM := $(GOMOD:.mod=.sum)
+DOCKER_IMAGE_TAG?=latest
+GO_CACHE_VOLUME_NAME?=gocache
+FIRECRACKER_CONTAINERD_TEST_IMAGE?=localhost/firecracker-containerd-test
 
 REVISION := $(shell git rev-parse HEAD)
 
-all: demux-snapshotter
+INTEG_TEST_SUFFIX := _Isolated
+INTEG_TESTNAMES=$(shell docker run --rm \
+	--network=host \
+	--volume $(CURDIR)/..:/src \
+	--volume $(GO_CACHE_VOLUME_NAME):/go \
+	--entrypoint=/bin/bash \
+	--workdir=/src/snapshotter \
+	$(FIRECRACKER_CONTAINERD_TEST_IMAGE):$(DOCKER_IMAGE_TAG) \
+	-c "go test -list . | sed '$$d' | grep $(INTEG_TEST_SUFFIX)")
+
+all: demux-snapshotter http-address-resolver
 
 demux-snapshotter: $(SOURCES) $(GOMOD) $(GOSUM)
 	go build $(EXTRAGOARGS) -ldflags "-X main.revision=$(REVISION)" -o $@
 
 http-address-resolver: $(SOURCES) $(GOMOD) $(GOSUM)
 	go build $(EXTRAGOARGS) -ldflags "-X main.revision=$(REVISION)" -o $@ internal/http_address_resolver.go
 
+install: demux-snapshotter http-address-resolver
+	install -D -o root -g root -m755 -t $(INSTALLROOT)/bin $^
+
 test:
 	go test ./... $(EXTRAGOARGS)
 
 integ-test:
+	$(MAKE) $(addprefix integ-test-,$(INTEG_TESTNAMES))
+
+integ-test-%: logs
+	docker run --rm -it \
+		--privileged \
+		--ipc=host \
+		--network=host \
+		--volume /dev:/dev \
+		--volume /run/udev/control:/run/udev/control \
+		--volume $(CURDIR)/logs:/var/log/firecracker-containerd-test \
+		--volume $(CURDIR)/..:/src \
+		--volume $(GO_CACHE_VOLUME_NAME):/go \
+		--env ENABLE_ISOLATED_TESTS=1 \
+		--env GOPROXY=direct \
+		--env GOSUMDB=off \
+		--env GO111MODULE=on \
+		--env NUMBER_OF_VMS=$(NUMBER_OF_VMS) \
+		--workdir="/src/snapshotter" \
+		--init \
+		$(FIRECRACKER_CONTAINERD_TEST_IMAGE):$(DOCKER_IMAGE_TAG) \
+		"go test $(EXTRAGOARGS) -run '^$(subst integ-test-,,$@)$$'"
+
+logs:
+	mkdir logs
 
 clean:
 	- rm -f demux-snapshotter
-	- rm -f http-resolver
+	- rm -f http-address-resolver
 
 distclean: clean
+	- rm -rf logs
 
-# Install is a noop here, for now.
-install:
-
-.PHONY: all install test clean distclean
+.PHONY: all demux-snapshotter http-address-resolver clean distclean install test integ-test
```

Comment:
https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html

> A phony target is one that is not really the name of a file; rather it is just a name for a recipe to be executed when you make an explicit request.

So demux-snapshotter and http-address-resolver shouldn't be listed here.

Reply:
Oh no my `make`-foo is slipping.